### PR TITLE
fix(component): improve contrast of text and buttons for night cards

### LINKED ISSUE #33: Improve contrast of text and buttons for night cards
**Describe the bug**
With fb129989a72084b6a679d54b22e1ca45798d7c24 the implementation of the background color of CardComponent was refactored. In this event the contrast optimizations for night cards weren't migrated (text and buttons).

**To Reproduce**

1. Choose "Nocturne".
2. Generate until a night card is shown.

**Expected behavior**
Display text and button in white for better contrast.

### PR DESCRIPTION
**Pull request by GagaMen**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit messages follow our guidelines
-   [x] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [x] Bugfix
-   [ ] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Other... Please describe:

## Closing Issue(s)

fix #33

## What is the current behavior?

Bad contrast of text and buttons for night cards.

## What is the new behavior?

Text and buttons for night cards are now white.

### CODE REVIEW

File: src/app/components/card/card.component.ts [hunk 1]
```diff
@@ -144,4 +151,33 @@ export class CardComponent {
 
         return '';
     }
+
+    private calculateBaseColor(): RGB {
```
Comment (TobiDimmel) [inline src/app/components/card/card.component.ts:155]: move behind `subtitleColor()`

File: src/app/components/card/card.component.ts [hunk 2]
```diff
@@ -83,25 +105,10 @@ export class CardComponent {
...
-            CardTypeId.Event,
-            CardTypeId.Ruins,
-            CardTypeId.Landmark,
-            CardTypeId.Project,
-            CardTypeId.Way,
-            CardTypeId.Trait,
-            CardTypeId.Ally,
-        ];
+    get background(): string {
         const orderedSecondaryCardTypes = [CardTypeId.Action, CardTypeId.Night];
```
Comment (TobiDimmel) [inline src/app/components/card/card.component.ts:109]: make it a field like `orderedPrimaryCardTypes` for more consistency

File: src/app/components/card/card.component.ts [hunk 3]
```diff
@@ -144,4 +151,33 @@ export class CardComponent {
...
     }
+
+    private calculateBaseColor(): RGB {
+        const containsTypeNight = this.card.types.some(
+            (cardType: CardType) => cardType.id === CardTypeId.Night,
+        );
+        const containsPrimaryType = this.card.types.some(
+            (cardType: CardType) =>
+                this.orderedPrimaryCardTypes.includes(cardType.id) ||
+                cardType.id === CardTypeId.Action,
```
Comment (TobiDimmel) [inline src/app/components/card/card.component.ts:162]: Could we just add `CardTypeId.Action` to `orderedPrimaryCardTypes`?

**Review by TobiDimmel — CHANGES_REQUESTED** [review]
see comments

File: src/app/components/card/card.component.ts (hunk 1 above)
Comment (GagaMen) [reply src/app/components/card/card.component.ts:167]: done

File: src/app/components/card/card.component.ts (hunk 2 above)
Comment (GagaMen) [reply src/app/components/card/card.component.ts]: done

File: src/app/components/card/card.component.ts (hunk 3 above)
Comment (GagaMen) [reply src/app/components/card/card.component.ts:174]: This unfortunately leads to problem with the calculation of the background color.

**Review by TobiDimmel — APPROVED** [review]
LGTM

**Pull request merged**